### DISCUSSION TITLE
Is my Papermill DD supposed to look like this?

### DISCUSSION THREAD

**Discussion by Spockrates** [OpenBB-finance/OpenBB#202] [Q&A] [answered]
New to Jupyter, so can't be sure, but I feel like my generated DD from the 'mill' menu is going awry.

See:

[MSFT_20210316_204627_due_diligence.pdf](https://github.com/DidierRLopes/GamestonkTerminal/files/6152971/MSFT_20210316_204627_due_diligence.pdf)

Something I'm missing? Thanks!

**Comment by aia** [comment, marked as answer]
Are you concerned with
```
Warning: Output is not a terminal (fd=None).
Warning: Input is not a terminal (fd=0).
```

**Reply by Spockrates** [reply]
Yes and moreover, whether or not any of those other sections should have generated a more "human readable" output.

**Reply by aia** [reply]
Sorry, with the exception of [Peter Norvig's notebooks](https://github.com/norvig/pytudes/blob/master/ipynb/Advent%20of%20Code.ipynb), pretty much all Jupyter looks like this.

![external-content duckduckgo com](https://user-images.githubusercontent.com/535247/111399469-ac542c00-8682-11eb-8fac-3de9d9c9acdd.jpg)

**Reply by Spockrates** [reply]
Ha! Gotcha. Initially when I saw I could generate PDF's of DD, I thought I'd be able to generate something I could share with laypersons over forums and such. Do you think this is worth creating a suggestion for or is this outside of the scope/purpose of the Papermill DD function?

**Reply by aia** [reply]
We could add another item to the mill menu. Do you want to suggest a notebook that would present information that you are looking for?

Personally for me the most important characteristics of the Jupyter notebook are:
1. Reproducibility - I want to be able to run and repeat evaluations that other people did
2. Clear chain of evaluations. I want to understand how the results were achieved.
3. Easy to change parameters here and there.

**Reply by Spockrates** [reply]
Perhaps a notebook isn't needed. Perhaps I could just go into the Mill menu and type something like:

`pdf -t AAPL income balance shrs ema rsi insider sec shorts`

And have that generate out a decent looking PDF report with those elements printed.

Is that something?

**Reply by aia** [reply]
The purpose of the notebook is to do some pre-calculations and establish a way to reproduce results.

PDFs printed today cannot be reproduced tomorrow.

Making financial decisions on information and calculations that cannot be reproduced is similar to reading DDs on WallStreetBets - un-reproducible studies are considered worthless in scientific method.

Pre-filled notebook is also just the beginning. I oftentimes edit notebooks to add other pieces of information, play with hyper parameters, or simply write my notes. At a later date I can always go back to my analysis, re-check it or just recall what I was thinking.

Machine Learning is all about running and re-running calculations with hyper parameter adjustments. Jupyter is the perfect vehicle for that workflow.

**Reply by Spockrates** [reply]
Sorry, I certainly didn't mean notebooks aren't needed in general. I was trying to gauge the value of a PDF DD option as an _additional_ feature not a replacement – and I was musing that for this additional feature, perhaps we wouldn't need as much functionality as the notebook. I agree with you that it doesn't replace the functionality of Jupyter.
